packages/core/core/src/services/document-service/repository.ts

```diff
@@ -254,20 +262,21 @@ export const createContentTypeRepository: RepositoryFactoryMethod = (uid) => {
   async function unpublish(opts = {} as any) {
     const { documentId, ...params } = opts;
 
-    const queryParams = await async.pipe(
+    const query = await async.pipe(
       i18n.defaultLocale(contentType),
-      i18n.multiLocaleToLookup(contentType)
+      i18n.multiLocaleToLookup(contentType),
+      transformParamsToQuery(uid),
+      (query) => assoc('where', { ...query.where, documentId, publishedAt: { $ne: null } }, query)
```

Comment:
will query.where always be an object here ?

Reply:
can it be something else? (if it's undefined the spread operator will work to I believe)

Reply:
if it went through the convertquery params it should be fine